Linear.lua

```diff
@@ -99,3 +99,9 @@ end
 
 -- we do not need to accumulate parameters when sharing
 Linear.sharedAccUpdateGradParameters = Linear.accUpdateGradParameters
+
+
+function Linear:__tostring__()
+  return 'nn.Linear' ..
```

Comment:
it would be better to do:
return torch.type(self) ..

so that if you ever subclass nn.Linear, the string returned will have the subclass name